osdep/timer-linux.c

```diff
@@ -35,11 +35,12 @@ void mp_sleep_ns(int64_t ns)
 uint64_t mp_raw_time_ns(void)
 {
     struct timespec tp = {0};
+    int ret = -1;
 #if defined(CLOCK_MONOTONIC_RAW)
-    clock_gettime(CLOCK_MONOTONIC_RAW, &tp);
-#else
-    timespec_get(&tp, TIME_UTC);
+    ret = clock_gettime(CLOCK_MONOTONIC_RAW, &tp);
 #endif
+    if (ret)
```

Comment:
nitpick I would fallback only on EINVAL, but it is fine too.

Reply:
I figure it can't hurt trying again on any failure. Although realistically if this fails, you're probably in really bad trouble anyway.